libraries/SPI/SPI.cpp

```diff
@@ -196,61 +199,203 @@ void ArduinoSPI::end()
 
-  for (auto const start = millis();
-       (SPI_EVENT_TRANSFER_COMPLETE != _spi_cb_event[_cb_event_idx]) && (millis() - start < 1000); )
-  {
-      __NOP();
-  }
-  if (SPI_EVENT_TRANSFER_ABORTED == _spi_cb_event[_cb_event_idx])
-  {
-      end();
-  }
+        for (auto const start = millis();
+            (SPI_EVENT_TRANSFER_COMPLETE != _spi_cb_event[_cb_event_idx]) && (millis() - start < 1000); )
+        {
+            __NOP();
+        }
+        if (SPI_EVENT_TRANSFER_ABORTED == _spi_cb_event[_cb_event_idx])
+        {
+            end();
+        }
+    }
+    else
+    {
+        if (buf != NULL) {
+            uint32_t *buffer32 = (uint32_t *) buf;
+            size_t ir = 0;
+            size_t it = 0;
+            size_t const n32 = count / 4U;
+            uint8_t const bytes_remaining = (uint8_t) (count & 3U);
+
+            if (n32 != 0U) {
+                _spi_ctrl.p_regs->SPCR_b.SPE = 0; /* disable SPI unit */
+                _spi_ctrl.p_regs->SPDCR = R_SPI0_SPDCR_SPLW_Msk; /* SPI word access */
+                _spi_ctrl.p_regs->SPCMD_b[0].SPB = 2; /* spi bit width = 32 */
+                _spi_ctrl.p_regs->SPCR_b.SPE = 1; /* enable SPI unit */
+
+                while ((it < 2U) && (it < n32)) {
+                    if (_spi_ctrl.p_regs->SPSR_b.SPTEF) {
+                        _spi_ctrl.p_regs->SPDR = buffer32[it];
+                        it++;
+                    }
+                }
+
+                while (it < n32) {
+                    if (_spi_ctrl.p_regs->SPSR_b.SPRF) {
+                        uint32_t tmp = _spi_ctrl.p_regs->SPDR;
+                        _spi_ctrl.p_regs->SPDR = buffer32[it];
+                        buffer32[ir] = tmp;
+                        ir++;
+                        it++;
+                    }
+                }
+
+                while (ir < n32) { /* collect the last word received */
+                    if (_spi_ctrl.p_regs->SPSR_b.SPRF) {
+                        uint32_t tmp = _spi_ctrl.p_regs->SPDR;
+                        buffer32[ir] = tmp;
+                        ir++;
+                    }
+                }
+
+                _spi_ctrl.p_regs->SPCR_b.SPE = 0; /* disable SPI unit */
+                _spi_ctrl.p_regs->SPDCR = R_SPI0_SPDCR_SPBYT_Msk; /* SPI byte access */
+                _spi_ctrl.p_regs->SPCMD_b[0].SPB = 7; /* spi bit width = 8 */
+                _spi_ctrl.p_regs->SPCR_b.SPE = 1; /* enable SPI unit */
+            }
+
+            /* send the remaining bytes with 8-bit transfers */
+            uint8_t *buffer = (uint8_t *) &buffer32[ir];
+
+            for (uint8_t index = 0; index < bytes_remaining; index++) {
+                _spi_ctrl.p_regs->SPDR_BY = buffer[index];
+                while (0U == _spi_ctrl.p_regs->SPSR_b.SPRF) {}
+                buffer[index] = _spi_ctrl.p_regs->SPDR_BY;
+            }
+        }
+    }
+}
+
+void ArduinoSPI::transfer(void *buf, void *rxbuf, size_t count)
```

Comment:
As much as I'd like to allow this method in, I can't.

Escalating to @facchinm .

Reply:
Hi @RudolphRiedel :coffee: :wave: 

Word has come down, no new APIs will be added, I therefore have to ask you to remove this function from this PR.

Please understand that it's difficult to add any API, as we'll need to add support across *all* official Arduino cores and either beg or support the major 3rd party cores in adjusting to this API extension as well. The Arduino ecosystem is strong, because this very small subset is shared across many platforms. Once every core does its own thing, the "Arduino" experience that allows easy porting of your sketch across many platforms will be lost.

We are open to modifications to the SPI class that would allow easy sub-classing, i.e. you could derive your own `SPIDMA` class and have all the fun you want, pending that the experience of users of the default library is not unduly burdened by such modifications.

Reply:
done, unfortunately
Really, at some point the API needs a number of updates.
For example a transfer(uint32) function as well.